schemas/logs/salesforce/insecure-external-assets.yml

```diff
@@ -0,0 +1,145 @@
+schema: Salesforce.InsecureExternalAssets
+version: 0
+referenceURL: https://developer.salesforce.com/docs/atlas.en-us.api.meta/api/sforce_api_objects_eventlogfile_insecureexternalassets.htm
+description: >-
+  Insecure External Assets events contain information about external assets.
+  External assets include images or videos accessed by users over an insecure HTTP protocol.
+  The event lists all your Salesforce pages that contain assets hosted insecurely on third-party sites
+  that users loaded with a Chrome, Firefox, Microsoft Edge, or Safari browser.
+parser:
+  csv:
+    delimiter: ","
+    hasHeader: true
+# Note field sequence is not verified and is likely different.
+fields:
+  - name: EVENT_TYPE
+    type: string
+    required: true
+    validate:
+      allow: ["InsecureExternalAssets"]
+    description: "The type of event. The value is always InsecureExternalAssets."
+  - name: ASSET_TYPE
+    type: string
+    required: true
+    description: >-
+      Type of insecure asset.
+      Possible Values: Base URI, Connect, Font,
+      Frame Ancestor: External page that embeds the Salesforce page in an iframe,
+      Frame, Image, Media, Object, Other, Plugin Types, Script, Style
+  - name: CLIENT_IP
+    required: false
+    type: string
+    indicators:
+      - ip
+    description: >-
+      The IP address of the client that’s using Salesforce services.
+      A Salesforce internal IP (such as a login from Salesforce Workbench
+      or AppExchange) is shown as "Salesforce.com IP".
+      For example: 10.0.0.1.
+  - name: CPU_TIME
+    required: true
+    type: bigint
+    description: >-
+      The CPU time in milliseconds used to complete the request.
+      This field indicates the amount of activity taking place in the app server layer.
+  - name: DOCUMENT_URI
+    required: true
```

Comment:
This is a schema for a CSV file... since it's CSV, I imagine that all these fields will be present at least with some empty value `""` or `-`. Does the `required` mean that the field should be present or that the field should have a non-empty value?

If it's the former, no need to change anything. 
If it's the latter, it would be better to limit the number of fields we mark as `required`. We'd like to have sufficient fields marked as `required` to avoid misclassifications, but we shouldn't be too strict and miss events. In the case of Salesforce, it seems that marking `EVENT_TYPE` as required would be enough?

Reply:
Indeed `required` means that a non-empty value should be present. We don't have many CSV parsers so a chance of overlap is small now, but `event_type: "Login"` alone seems that could be quite common in the security context. I will revise the required fields though and reduce them to only a small subset (probably 2-3 fields should suffice). Thanks!

Reply:
Required fields reduced to a minimum.